rust/lance-datafusion/src/logical_expr.rs

```diff
@@ -91,6 +91,33 @@ pub fn resolve_column_type(expr: &Expr, schema: &Schema) -> Option<DataType> {
 /// - *schema*: lance schema.
 pub fn resolve_expr(expr: &Expr, schema: &Schema) -> Result<Expr> {
     match expr {
+        Expr::Between(Between {
+            expr: inner_expr,
+            low,
+            high,
+            negated,
+        }) => {
+            if let Some(inner_expr_type) = resolve_column_type(inner_expr.as_ref(), schema) {
+                let low = if matches!(low.as_ref(), Expr::Literal(_)) {
+                    Box::new(resolve_value(low.as_ref(), &inner_expr_type)?)
+                } else {
+                    low.clone()
+                };
+                let high = if matches!(high.as_ref(), Expr::Literal(_)) {
+                    Box::new(resolve_value(high.as_ref(), &inner_expr_type)?)
+                } else {
+                    high.clone()
+                };
```

Comment:
Is it possible to use something more generic like `let low = coerce_expr(left.as_ref(), &inner_expr_type)?;`

(note: I don't know off the top of my head if this works or not)